crates/edr_solidity/src/contracts_identifier/radix_tree.rs

```diff
@@ -0,0 +1,540 @@
+use std::collections::BTreeMap;
+
+use revm_primitives::Bytes;
+
+#[derive(Debug)]
+pub struct RadixNode {
+    content: Bytes,
+    is_present: bool,
```

Comment:
Why not make this an `Option<Bytes>`?

Reply:
Bytes is never missing, `is_present` is about that node corresponding to an actual word or not.

For example, let's say you add the words `ab` and `ac`. The tree would look like this:

```
 root
  |
  a
 / \
b   c
```

In this case the node that has `a` is not present, because it doesn't correspond to an actual word. If you then add the word `a`, then the only change to the tree is that the `is_present` field of that node becomes `true`. Does that make sense?

Reply:
And the `content` would then be:

a: a
b: ab
c: ac

?

Reply:
No, the content is not complete, it would simply be:

a: a
b: b
c: c

The image in [this article](https://en.wikipedia.org/wiki/Radix_tree) explains it better, although it doesn't include the `is_present` flag.